File: RunWear-iOS/RunWear/Services/ClothingThumbnailService.swift
import Foundation

/// Service for generating clothing thumbnail URLs.
///
/// Thumbnails are AI-generated product images stored in Supabase Storage.
/// 52 total images: 26 clothing items × 2 genders (male/female).
///
/// Style: Dark gray/charcoal clothing on pure black background,
/// professional product photography aesthetic.
class ClothingThumbnailService {
    static let shared = ClothingThumbnailService()

    private let baseURL = "https://ebicqznlcjbqcukjfzcf.supabase.co/storage/v1/object/public/clothing-thumbnails"

    private init() {}

    /// Map clothing item names to thumbnail file keys.
    /// Keys use lowercase with hyphens (e.g., "Tank Top" → "tank-top").
    private let thumbnailKeys: [String: String] = [
        // Tops - Base Layer
        "Tank Top": "tank-top",
        "Short Sleeve Shirt": "short-sleeve",
        "Long Sleeve Shirt": "long-sleeve-light",
        "Thermal Long Sleeve": "thermal-long-sleeve",
        "Base Layer + Jacket": "thermal-long-sleeve",
        "Double Layer Top": "insulated-jacket",
        "Full Thermal System": "insulated-jacket",

        // Tops - Outer Layer
        "Light Vest": "light-vest",
        "Windbreaker": "windbreaker",
        "Light Jacket": "light-jacket",
        "Rain Jacket": "rain-jacket",
        "Insulated Jacket": "insulated-jacket",

        // Bottoms
        "Split Shorts": "short-shorts",
        "Running Shorts": "running-shorts",
        "Running Tights": "light-tights",
        "Thermal Tights": "thermal-tights",
        "Insulated Tights": "thermal-tights",
        "Wind-Proof Pants": "thermal-tights",

        // Head
        "Visor": "visor",
        "Running Cap": "running-cap",
        "Headband": "headband",
        "Ear Warmer": "headband",
        "Light Beanie": "light-beanie",
        "Running Beanie": "light-beanie",
        "Thermal Beanie": "thermal-beanie",
        "Balaclava": "balaclava",
        "Full Balaclava": "balaclava",

        // Hands
        "Light Gloves": "light-gloves",
        "Running Gloves": "thermal-gloves",
        "Thermal Gloves": "thermal-gloves",
        "Heavy Gloves": "thermal-gloves",
        "Mittens": "mittens",
        "Insulated Mittens": "mittens",

        // Accessories
        "Sunglasses": "sunglasses",
        "Sunscreen": "sunscreen",
        "Reflective Gear": "reflective-gear",
        "Neck Gaiter": "neck-gaiter",
        "Arm Sleeves": "light-gloves", // Use gloves thumbnail as proxy
        "Hand Warmers": "thermal-gloves" // Use gloves thumbnail as proxy
    ]

    /// Get the thumbnail URL for a clothing item based on gender preference.
    ///
    /// - Parameters:
    ///   - item: The clothing item
    ///   - gender: The user's gender preference (.unisex defaults to male)
    /// - Returns: The full URL to the thumbnail image, or nil if no thumbnail exists
    func getThumbnailURL(for item: ClothingItem, gender: GenderPreference) -> URL? {
        guard let thumbnailKey = thumbnailKeys[item.name] else { return nil }

        let genderSuffix: String
        switch gender {
        case .female:
            genderSuffix = "female"
        case .male, .unisex:
            genderSuffix = "male"
        }

        return URL(string: "\(baseURL)/\(thumbnailKey)-\(genderSuffix).webp")
    }

    /// Check if a clothing item has a thumbnail available.
    func hasThumbnail(for item: ClothingItem) -> Bool {
        return thumbnailKeys[item.name] != nil
    }
}

File: RunWear-android/shared/src/main/java/com/runwear/shared/domain/service/ClothingThumbnailService.kt
package com.runwear.shared.domain.service

import com.runwear.shared.domain.model.ClothingItem
import com.runwear.shared.domain.model.GenderPreference

/**
 * Service for generating clothing thumbnail URLs.
 *
 * Thumbnails are AI-generated product images stored in Supabase Storage.
 * 52 total images: 26 clothing items × 2 genders (male/female).
 *
 * Style: Dark gray/charcoal clothing on pure black background,
 * professional product photography aesthetic.
 */
object ClothingThumbnailService {

    private const val BASE_URL = "https://ebicqznlcjbqcukjfzcf.supabase.co/storage/v1/object/public/clothing-thumbnails"

    /**
     * Map ClothingItem enum names to thumbnail file keys.
     * Keys use lowercase with hyphens (e.g., TANK_TOP → tank-top).
     */
    private val THUMBNAIL_KEYS: Map<ClothingItem, String> = mapOf(
        // Tops - Base Layer
        ClothingItem.TANK_TOP to "tank-top",
        ClothingItem.SHORT_SLEEVE to "short-sleeve",
        ClothingItem.LONG_SLEEVE_LIGHT to "long-sleeve-light",
        ClothingItem.LONG_SLEEVE_THERMAL to "thermal-long-sleeve",

        // Tops - Mid Layer
        ClothingItem.QUARTER_ZIP to "quarter-zip",
        ClothingItem.FLEECE_PULLOVER to "fleece-pullover",
        ClothingItem.HALF_ZIP_THERMAL to "half-zip-thermal",

        // Tops - Outer Layer
        ClothingItem.LIGHT_VEST to "light-vest",
        ClothingItem.WINDBREAKER to "windbreaker",
        ClothingItem.LIGHT_JACKET to "light-jacket",
        ClothingItem.RAIN_JACKET to "rain-jacket",
        ClothingItem.INSULATED_JACKET to "insulated-jacket",

        // Bottoms
        ClothingItem.SHORT_SHORTS to "short-shorts",
        ClothingItem.SHORTS to "running-shorts",
        ClothingItem.CAPRIS to "light-tights", // Capris use light-tights thumbnail
        ClothingItem.LIGHT_TIGHTS to "light-tights",
        ClothingItem.THERMAL_TIGHTS to "thermal-tights",

        // Head
        ClothingItem.VISOR to "visor",
        ClothingItem.BASEBALL_CAP to "running-cap",
        ClothingItem.HEADBAND to "headband",
        ClothingItem.LIGHT_BEANIE to "light-beanie",
        ClothingItem.THERMAL_BEANIE to "thermal-beanie",
        ClothingItem.BALACLAVA to "balaclava",

        // Hands
        ClothingItem.LIGHT_GLOVES to "light-gloves",
        ClothingItem.THERMAL_GLOVES to "thermal-gloves",
        ClothingItem.MITTENS to "mittens",

        // Accessories
        ClothingItem.SUNGLASSES to "sunglasses",
        ClothingItem.SUNSCREEN to "sunscreen",
        ClothingItem.REFLECTIVE_GEAR to "reflective-gear",
        ClothingItem.NECK_GAITER to "neck-gaiter"
    )

    /**
     * Get the thumbnail URL for a clothing item based on gender preference.
     *
     * @param item The clothing item
     * @param gender The user's gender preference (UNISEX defaults to male)
     * @return The full URL to the thumbnail image, or null if no thumbnail exists
     */
    fun getThumbnailUrl(item: ClothingItem, gender: GenderPreference): String? {
        val thumbnailKey = THUMBNAIL_KEYS[item] ?: return null
        val genderSuffix = when (gender) {
            GenderPreference.FEMALE -> "female"
            else -> "male" // MALE and UNISEX both use male thumbnails
        }
        return "$BASE_URL/$thumbnailKey-$genderSuffix.webp"
    }

    /**
     * Check if a clothing item has a thumbnail available.
     */
    fun hasThumbnail(item: ClothingItem): Boolean {
        return THUMBNAIL_KEYS.containsKey(item)
    }
}
